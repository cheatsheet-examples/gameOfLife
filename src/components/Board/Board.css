.Board {
    position: relative;
    margin: 0 auto;
    background-color: rgb(18, 56, 224);
    background-image:
        linear-gradient(#333 1px, transparent 1px),
        linear-gradient(90deg, #333 1px, transparent 1px);
}