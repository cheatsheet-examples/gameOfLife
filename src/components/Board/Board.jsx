import React from 'react';
import PropTypes from 'prop-types';
import Cell from '../Cell/Cell';
import './Board.css';

const Board = ({ handleSingleClick, width, height, cellSize, cells, boardRef }) => (

  <div className="Board"
    style={{ width: width, height: height, backgroundSize: `${cellSize}px ${cellSize}px` }}
    onClick={handleSingleClick}
    ref={boardRef}
  >
    {cells.map(cell => (
      <Cell key={`${cell.x},${cell.y}`} x={cell.x} y={cell.y} cellSize={cellSize} />
    ))}
  </div>
);

Board.propTypes = {
  handleSingleClick: PropTypes.func.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  cellSize: PropTypes.number,
  cells: PropTypes.array.isRequired,
  boardRef: PropTypes.func,
};

Board.defaultProps = {
  cells: [],
};

export default Board;