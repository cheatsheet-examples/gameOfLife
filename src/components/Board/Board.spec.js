import React from 'react';
import { shallow } from 'enzyme';
import Board from './Board';

const MOCK_CELL_SIZE = 20;
const MOCK_HANDLE_CLICK = jest.fn();
const MOCK_HEIGHT = 600;
const MOCK_WIDTH = 800;
const MOCK_CELLS = [];
const MOCK_BOARDREF = jest.fn();

describe('Board', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Board handleSingleClick={MOCK_HANDLE_CLICK} width={MOCK_HEIGHT} height={MOCK_WIDTH}
        cellSize={MOCK_CELL_SIZE} cells={MOCK_CELLS} boardRef={MOCK_BOARDREF} />
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });
});
