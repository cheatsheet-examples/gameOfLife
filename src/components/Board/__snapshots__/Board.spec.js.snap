// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render correctly 1`] = `
<div
  className="Board"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundSize": "20px 20px",
      "height": 800,
      "width": 600,
    }
  }
/>
`;
